pkg/cmd/cli/secrets/source.go

```diff
@@ -0,0 +1,230 @@
+)
+
+const (
+	CreateSourceSecretRecommendedCommandName = "new-sourcecfg"
+
+	createSourceSecretLong = `
+Create a new source secret
+
+Source secrets are used to authenticate against SCM servers.
+
+When creating applications, you may have a SCM server that requires authentication. In order for the
+nodes to clone source code on your behalf, they have to have the credentials. You can provide this
+information by creating a sourcecfg secret and attaching it to your service account.`
+
+	createSourceSecretExample = `
+If your source authentication method requires only username and password or token, add it by using:
+
+  $ %[1]s SECRET_NAME --username=USERNAME --password=PASSWORD
+
+If your source authentication method requires also a custom CA certificate, add it by using:
+
+  $ %[1]s SECRET_NAME --username=USERNAME --password=PASSWORD --ca-cert=FILENAME
+
+If your source authentication method requires private ssh key, add it by using:
+
+  $ %[1]s SECRET_NAME --ssh-privatekey=FILENAME
+
+If you do already have a .gitconfig file needed for authentication, you can create a gitconfig secret by using:
+
+  $ %[2]s SECRET_NAME path/to/.gitconfig`
+
+	optionGroupExclusivityError = `must provide only one kind of option group.
+  Basic authentication group flags: --username, --password, --ca-cert
+  SSH authentication group flags: --ssh-privatekey
+`
+)
+
+type CreateSourceSecretOptions struct {
+	SecretNamespace string
+	SecretName      string
+	Username        string
+	Password        string
+	PrivateKeyPath  string
+	SourceURL       string
+	CertificatePath string
+	GitConfigPath   string
+
+	PromptForPassword bool
+
+	Reader io.Reader
+	Out    io.Writer
+
+	SecretsInterface client.SecretsInterface
+}
+
+func NewCmdCreateSourceSecret(name, fullName string, f *kcmdutil.Factory, reader io.Reader, out io.Writer, newSecretFullName, ocEditFullName string) *cobra.Command {
+	o := &CreateSourceSecretOptions{
+		Out:    out,
+		Reader: reader,
+	}
+
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s SECRET_NAME (--username=USERNAME --password=PASSWORD --ca-cert=FILENAME | --ssh-privatekey=FILENAME) [--gitconfig=FILENAME]", name),
+		Short:   "Create a new source secret",
+		Long:    createSourceSecretLong,
+		Example: fmt.Sprintf(createSourceSecretExample, fullName, newSecretFullName, ocEditFullName),
+		Run: func(c *cobra.Command, args []string) {
+			if err := o.Complete(f, args); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			if err := o.Validate(); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			if len(kcmdutil.GetFlagString(c, "output")) != 0 {
+				secret, err := o.MakeSourceSecret()
+				kcmdutil.CheckErr(err)
+
+				kcmdutil.CheckErr(f.PrintObject(c, secret, out))
+				return
+			}
+
+			if err := o.CreateSourceSecret(); err != nil {
+				kcmdutil.CheckErr(err)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&o.Username, "username", "", "username for Git authentication")
```

Comment:
Yep, sorry for that ... was looking into dockercfg.go as an example and didn't noticed there is it wrong. :P